python/rpdk/java/templates/init/guided_aws/StubUpdateHandler.java

```diff
@@ -26,126 +25,88 @@
-                    .makeServiceCall(this::updateResource)
+                    .makeServiceCall((awsRequest, client) -> {
+                        AwsResponse awsResponse = null;
+                        try {
+
+                            // TODO: put your update resource code here
+
+                        } catch (final AwsServiceException e) {
+                            /*
+                            * While the handler contract states that the handler must always return a progress event,
+                            * you may throw any instance of BaseHandlerException, as the wrapper map it to a progress event.
+                            * Each BaseHandlerException maps to a specific error code, and you should map service exceptions as closely as possible
+                            * to more specific error codes
+                            */
+                            throw new CfnGeneralServiceException(ResourceModel.TYPE_NAME, e);
+                        }
+
+                        logger.log(String.format("%s has successfully been updated.", ResourceModel.TYPE_NAME));
+                        return awsResponse;
+                    })
 
                     // STEP 1.3 [TODO: stabilize step is not necessarily required but typically involves describing the resource until it is in a certain status, though it can take many forms]
                     // stabilization step may or may not be needed after each API call
                     // for more information -> https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html
-                    .stabilize(this::stabilizedOnFirstUpdate)
+                    .stabilize((awsRequest, awsResponse, client, model, context) -> {
+                        // TODO: put your stabilization code here
+
+                        final boolean stabilized = true;
+
+                        logger.log(String.format("%s [%s] update has stabilized: %s", ResourceModel.TYPE_NAME, model.getPrimaryIdentifier(), stabilized));
+                        return stabilized;
+                    })
                     .progress())
 
             // If your resource is provisioned through multiple API calls, then the following pattern is required (and might take as many postUpdate callbacks as necessary)
             // STEP 2 [second update/stabilize progress chain]
             .then(progress ->
                     // STEP 2.0 [initialize a proxy context]
-                    proxy.initiate("{{ call_graph }}::{{ operation }}", proxyClient, model, callbackContext)
+                    // If your resource is provisioned through multiple API calls, you will need to apply each subsequent update
+                    // step in a discrete call/stabilize chain to ensure the entire resource is provisioned as intended.
+                    proxy.initiate("{{ call_graph }}::{{ operation }}::second", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
```

Comment:
```suggestion
                    proxy.initiate("{{ call_graph }}::{{ operation }}::stabilize", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
```

Reply:
same here, it's not really for stabilization. it's for the second update call as a rough example